tests/admin_widgets/tests.py

```diff
@@ -762,6 +766,39 @@ def test_render_fk_as_pk_model(self):
             'class="related-lookup" id="lookup_id_test" title="Lookup"></a>',
         )
 
+    def test_quoted_pk_value_on_url(self):
+        house = House.objects.create(name="?a=b")
+        rel = Room._meta.get_field("house").remote_field
+        w = widgets.ForeignKeyRawIdWidget(rel, widget_admin_site)
+        rendered = w.render("test", house.pk, attrs={})
+
+        expected_url = "/admin_widgets/house/%s/change/" % quote(house.pk)
+
+        self.assertIn(expected_url, rendered)
+        self.assertHTMLEqual(
+            w.render("test", house.pk, attrs={}),
+            '<input class="vForeignKeyRawIdAdminField" name="test" type="text" '
+            'value="%(val)s"> '
+            '<a href="/admin_widgets/house/?_to_field=name" class="related-lookup" '
+            'id="lookup_id_test" title="Lookup"></a> '
+            '<strong><a href="/admin_widgets/house/%(pk)s/change/">%(val)s</a></strong>'
+            % {"val": house.pk, "pk": quote(house.pk)},
+        )
+
+    def test_foreign_key_raw_id_widget_renders_quoted_pk_in_change_url(self):
+        house = House.objects.create(name="_40")
+        rel = Room._meta.get_field("house").remote_field
+        w = widgets.ForeignKeyRawIdWidget(rel, widget_admin_site)
+
+        # apply quote function to primary key value
+        pk_quoted = quote(str(house.pk))
+
+        # render the widget
+        rendered = w.render("test", house.pk, attrs={})
+
+        # check that the primary key is properly quoted in the rendered HTML
+        self.assertIn(f'href="/admin_widgets/house/{pk_quoted}/change/"', rendered)
```

Comment:
All comments are unnecessary as they don't add more explanation/
```suggestion
        pk_quoted = quote(str(house.pk))
        rendered = w.render("test", house.pk, attrs={})
        self.assertIn(f'href="/admin_widgets/house/{pk_quoted}/change/"', rendered)
```